packages/react-aria-components/src/RadioGroup.tsx

```diff
@@ -133,11 +135,23 @@ function RadioGroup(props: RadioGroupProps, ref: ForwardedRef<HTMLDivElement>) {
     defaultClassName: 'react-aria-RadioGroup'
   });
 
+  useImperativeHandle<HTMLDivElement|null, HTMLDivElement|null>(ref, () => {
+    if (domRef.current) {
+      domRef.current.focus = () => {
+        if (domRef.current) {
+          let walker = getFocusableTreeWalker(domRef.current, {tabbable: true});
+          (walker.firstChild() as HTMLElement).focus();
```

Comment:
@sookmax After taking a brief look at [`useFocusVisible()`](https://github.com/adobe/react-spectrum/blob/1697a7fef3516f7a194f43b75edb858589ef6535/packages/%40react-aria/interactions/src/useFocusVisible.ts#L302), i think it might be okay to dispatch a keyboard event manually before the focus occurs. This triggers isFocusVisible to be activated, but might be good to get more input here from core members.

```tsx
let radio = getFocusableTreeWalker(domRef.current, {tabbable: true}).firstChild();

radio.dispatchEvent(new KeyboardEvent('keydown'));
radio.focus();
```

Reply:
One consideration would be if native controls do this across all three browsers. Someone could check this with a jsfiddle or something pretty quickly.
Otherwise, I'd prefer to match Checkbox and Switch and avoid firing an extra event.

Reply:
By all three you mean Chrome, FireFox, and Safari?

If so, I checked them with [this jsfiddle](https://jsfiddle.net/sookmax/f0wqdj1m/1), and Safari is the only one that doesn't add a visual indicator properly.

### Chrome
<img width="596" alt="native-radio-chrome" src="https://github.com/adobe/react-spectrum/assets/71210554/fa5f85b5-45cf-49e5-8494-e34bd4e09657">

### FireFox
<img width="644" alt="native-radio-firefox" src="https://github.com/adobe/react-spectrum/assets/71210554/7af9fa9e-da98-4501-8e8f-0d43c965a78f">

### Safari
<img width="644" alt="native-radio-safari" src="https://github.com/adobe/react-spectrum/assets/71210554/06a0bea8-e102-42b4-8b20-0f1a41486ee8">

---
Safari is a little weird in that when you open the inspector and select one of the radio inputs and apply both pseudo-classes `:focus` and `:focus-visible`, you can see the styled outline being applied, and _from that moment on_.. if you re-run the fiddle, it will start to show the outline correctly. _However_, if you refresh Safari, it stops working. (it's even weirder to explain this in plain English 😂)

<img width="1470" alt="Screenshot 2024-03-21 at 11 40 15 PM" src="https://github.com/adobe/react-spectrum/assets/71210554/d2ac28a3-b9a3-482d-a2af-cb962bce376b">

Reply:
thank you very much for checking that, looks like we'll have to decide if we follow this

Reply:
If I have a say, then I'd like to support this feature unless firing a fake keyboard event incurs a severe side effect. If consistency with Checkbox and Switch is a concern, then I can also take a look at them as well.

I think having the focus-visible state when the user calls ref.focus() enhances the user experience after all.

Reply:
Hey, I finally got some time to look at this a little more. Thanks again for the jsfiddle example. Unfortunately, the behavior changes depending on when/how you call the 'focus'. See this fiddle https://jsfiddle.net/re7d3gum/2/ where focus is called after clicking a button. None of the browsers show focus as visible.

Focus is visible if you press the button with keyboard across all three browsers as well, which makes sense.

Therefore, I think we do not need to fire an extra keyboard event to show the focus visible.

Reply:
Ha, it's interesting how the browser chooses not to give a focus indicator when a button is interacted with a mouse! Thanks for verifying this.

But I still cannot help wondering about a way for the user to style the focused (via `ref.focus()`) radio.. well I guess, at least `ref.focus()` gives the `<label>` `data-focused` state, but if you try to style a focused radio generically like this:
```css
.radio {
  &:before {
      content: '';
      display: block;
      width: 1.286rem;
      height: 1.286rem;
      box-sizing: border-box;
      border: 0.143rem solid var(--deselected-color);
      background: var(--background-color);
      border-radius: 1.286rem;
      transition: all 200ms;
    }
  
  &[data-focused]:before {
      box-shadow: 0 0 0 2px canvas, 0 0 0 4px var(--selected-color);
    }
}
```

Then (other than the intended use case of when `ref.focus()` is called) each radio will get focused every time it is clicked by a pointing device.

Would there be a better way to target (stylistically) this specific use case (`ref.focus()`)?